Add json_array_parser parser and assign_keys transformer (#30644)
open-telemetry/opentelemetry-collector-contrib: pkg/stanza/docs/operators/assign_keys.md

> ## `assign_keys` operator

Author: Done

The `assign_keys` assigns keys from the configuration to an input list. the output is a map containing these key-value pairs

### Configuration Fields

| Field      | Default          | Description |
| ---        | ---              | ---         |
| `id`       | `assign_keys`        | A unique identifier for the operator. |
| `output`   | Next in pipeline | The connected operator(s) that will receive all outbound entries. |
| `field`    | required         | The [field](../types/field.md) to assign keys to. |

> | `keys`    | required         | The list of strings to be used as the keys to the input list's values. It's length is expected to be equal to the length of the values list from field. In case there is a mismatch, assign_keys will return an error and the field will remain untouched|

Reviewer: Thanks for adding the note, just compressing this a bit.
```suggestion
| `keys`    | required         | The list of strings to be used as the keys to the input list's values. Its length is expected to be equal to the length of the values list from field. In case there is a mismatch, an error will result. |
```
